plugins/permission-react/src/apis/PermissionApi.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AuthorizeRequest,
+  AuthorizeResponse,
+} from '@backstage/plugin-permission-common';
+import { ApiRef, createApiRef } from '@backstage/core-plugin-api';
+
+/**
+ * This API is used by various frontend utilities that allow developers to implement authorization wihtin their frontend
+ * plugins. A plugin developer will likely not have to interact with this API or its implementations directly, but
+ * rather with the aforementioned utility components/hooks.
+ * @public
+ */
+export type PermissionApi = {
+  authorize(
+    requests: Array<AuthorizeRequest>,
```

Comment:
What do you think about having this API handle a single request at a time? Seems like that would provide for a tiny bit better DX tbh to not have to deal with the array wrapping and unwrapping. Then we'd move the batching of permission queries to an internal optimization of the API rather than something the user needs to do. Starting point there could be to wait just an ms or two and then send off all pending requests.

Reply:
I like that idea. @mtlewis thoughts?

Reply:
Sounds good to me!

Reply:
One question is whether we want to propagate this change to the PermissionClient's `authorize` method as well (and subsequently be equipped to handle batching there in the future). That seems like the logical move, though that introduces changes we'll have to make in permission-common which would be outside the scope of this PR. Happy to follow up with a separate PR for that if we think it makes sense. Either way, I've left the singleton -> array conversion layer inside of the `IdentityPermissionApi` implementation for now.